assets/js/components/ChecksCatalog/ChecksCatalog.jsx

```diff
@@ -24,7 +24,9 @@ const updatedProvider = {
 };
 
 const buildUpdateCatalogAction = (provider) => {
-  const payload = checkProviderExists(provider) ? { provider } : {};
+  const payload = checkProviderExists(provider)
```

Comment:
What if we choose `All` as provider filtering. This would return `false` in `checkProviderExists` and we wouldn't filter by `resource_type`, right?
It looks like unfinished. 
I don't know if we should start filtering here behind the scenes before knowing how the user will filter

Reply:
Correct, when filtering by `All` we don't get anything in the qs, meaning give me everything (wanda does not apply any filtering if an empty `env` was provided)

I believe this is how far we can get by now.

Reply:
>  why the filtering doesn't work without the resource_type?
Because the when expression is just one expression ie `when: env.resource_type == "cluster" && env.provider !in ["gcp", "aws"]` and if no resource type is in the env or no provider you see it fails.

> If I were a user and would like to get all Azure checks, why I'm forced to filter by resource type?
I agree. That's the limitation of having everything in the when condition.